libs/contrib/Data/String/Parser.idr

```diff
@@ -290,3 +314,47 @@ integer : Monad m => ParseT m Integer
 integer = do minus <- succeeds (char '-')
              x <- some digit
              pure $ if minus then (intFromDigits x)*(-1) else intFromDigits x
+
+
+||| Parse repeated instances of at least one `p`, separated by `s`,
+||| returning a list of successes.
+|||
+||| @ p the parser for items
+||| @ s the parser for separators
+export
+covering
+sepBy1 : Monad m => (p : ParseT m a)
+                 -> (s : ParseT m b)
+                 -> ParseT m (List a)
```

Comment:
Consider returning a `List1` given that `sepBy1` returns a non-empty list.

Reply:
Good call, updated. It made `sepBy` a bit of fun to implement as i had to call `forget`, but I think I got it right.